.travis.yml

```diff
@@ -29,97 +24,55 @@ env:
-    ; do
-      if \
-          wget -t 3 -O "${bazel_binary}" "${url}" &&
-          shasum -a 256 --check "${bazel_checksum_file}"; then
-        break
-      else
-        rm -f "${bazel_binary}"
-      fi
-    done &&
-    rm "${bazel_checksum_file}" &&
-    [ -f "${bazel_binary}" ]
-  - chmod +x "${bazel_binary}"
-  - sudo mv "${bazel_binary}" /usr/local/bin/bazel
-
-  # Fix Boto and Travis issue https://github.com/travis-ci/travis-ci/issues/7940
-  - sudo rm -f /etc/boto.cfg
-
-  # Storing build artifacts in this directory helps Travis cache them. This
-  # will sometimes cut latency in half, when we're lucky.
-  - echo "startup --output_base=${HOME}/.bazel-output-base" >>~/.bazelrc
-
-  # Travis Trusty Sudo GCE VMs have 2 cores and 7.5 GB RAM. These settings
-  # help Bazel go faster and not OOM the system.
-  - echo "startup --host_jvm_args=-Xms500m" >>~/.bazelrc
-  - echo "startup --host_jvm_args=-Xmx500m" >>~/.bazelrc
-  - echo "startup --host_jvm_args=-XX:-UseParallelGC" >>~/.bazelrc
-  - echo "build --local_resources=400,2,1.0" >>~/.bazelrc
-  - echo "build --worker_max_instances=2" >>~/.bazelrc
-
-  # Make Bazel as strict as possible, so TensorBoard will build correctly
-  # for users, regardless of their Bazel configuration.
-  - echo "build --worker_verbose" >>~/.bazelrc
-  - echo "build --worker_sandboxing" >>~/.bazelrc
-  - echo "build --spawn_strategy=sandboxed" >>~/.bazelrc
-  - echo "build --genrule_strategy=sandboxed" >>~/.bazelrc
-  - echo "test --test_verbose_timeout_warnings" >>~/.bazelrc
-
-  # It's helpful to see the errors on failure.
-  - echo "build --verbose_failures" >>~/.bazelrc
-  - echo "test --test_output=errors" >>~/.bazelrc
-
-  # We need to pass the PATH from our virtualenv down into our tests,
-  # which is non-hermetic and so disabled by default in Bazel 0.21.0+.
-  - echo "test --action_env=PATH" >>~/.bazelrc
+  - elapsed() { TZ=UTC printf "Time %(%T)T %s\n" "$SECONDS" "$@"; }
+  - elapsed "before_install"
+  - ci/download_bazel.sh "${BAZEL}" "${BAZEL_SHA256SUM}" ~/bazel
+  - sudo mv ~/bazel /usr/local/bin/bazel
+  - cp ci/bazelrc ~/.bazelrc
+  - elapsed "before_install (done)"
 
 install:
-  - pip install boto3==1.9.86
+  - elapsed "install"
+  # Lint check deps.
   - pip install flake8==3.5.0
-  - pip install futures==3.1.1
-  - pip install grpcio==1.6.3
-  - pip install moto==1.3.7
   - pip install yamllint==1.5.0
+  # TensorBoard deps.
+  - pip install futures==3.1.1
+  - pip install grpcio==1.0
+  # Uninstall older Travis numpy to avoid upgrade-in-place issues.
+  - pip uninstall -y numpy
   - |
     # Install TensorFlow if requested
     if [ -n "${TF_VERSION_ID}" ]; then
       pip install -I "${TF_VERSION_ID}"
     else
-      # Requirements typically found through TensorFlow
-      pip install "absl-py>=0.7.0"
-      pip install "numpy<2.0,>=1.14.5"
+      # Requirements typically found through TensorFlow.
+      pip install "absl-py>=0.7.0" \
```

Comment:
Ooh, good catch. Would be nice to have some kind of [`--chain-lint`][1].

[1]: https://github.com/git/git/blob/aa25c82427ae70aebf3b8f970f2afd54e9a2a8c6/t/README#L188-L196